frontend/src/components/Section.js

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import marked from "marked";
+
+class Section extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = Object.assign({}, {editMode: false}, this.stateFromSection(props.section))
+
+    this.toggleEdit = this.toggleEdit.bind(this)
+    this.toggleEdit = this.toggleEdit.bind(this)
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.saveContentToServer = this.saveContentToServer.bind(this)
+  }
+
+  stateFromSection(section) {
+    return {
+      sectionId: section.id,
+      content: section.content,
+      isAdminOnly: section.is_admin_only,
+      priorContent: section.content,
+      priorIsAdminOnly: section.is_admin_only,
+    }
+  }
+
+  handleChange(event) {
+    if (event.target.name === 'content') {
+      this.setState({content: event.target.value})
+    } else if ( event.target.name === 'isAdminOnly') {
+      this.setState({isAdminOnly: event.target.checked})
+    }
+  }
+
+  handleSubmit(event) {
+    alert('a Thing was submitted' + this.state.value);
+    event.preventDefault();
+  }
+
+  saveContentToServer() {
+     const requestOptions = {
+       method: 'POST',
+       headers: { 'Content-Type': 'application/json' },
+       body: JSON.stringify({
+            id: this.state.sectionId,
+            content: this.state.content,
+            wiki_id: this.props.wikiId,
+            page_id: this.props.pageId,
+            is_admin_only: this.state.isAdminOnly,
+         })
+     };
+     fetch(`http://localhost:8000/api/w/${this.props.wikiId}/p/${this.props.pageId}/s/${this.state.sectionId}`, requestOptions)
+        .then(response => response.json())
+        .then(section => {
+          this.setState(Object.assign({}, { editMode: false }, this.stateFromSection(section)))
+        });
+  }
+
+  toggleEdit() {
+    if (this.state.editMode)  {
+      this.setState({editMode: false, isAdminOnly: this.state.priorIsAdminOnly})
+    } else {
+      this.setState({editMode: true})
+    }
+  }
+
+  markdownContent(thing) {
+    return { __html: marked(this.state.priorContent, {sanitize: true}) };
```

Comment:
I think this opens you up to a cross-site scripting vulnerability. To quote [Marked's docs], "Marked does not sanitize the output HTML. Please use a sanitize library, like DOMPurify (recommended), sanitize-html or insane on the output HTML!"

I don't have an obvious answer, but I'd consider switching to [Remarkable](https://github.com/jonschlinkert/remarkable) because that's what's used on React's homepage, and it looks like it disables some of this stuff by default. But I'm not sure.

[0]: https://marked.js.org/#usage

Reply:
Yeah, this is a long-standing issue :D For now I'm relying on verrrry limited users even for reads.